### PR TITLE
Fix prefetched data loading

### PR DESCRIPTION
**Pull request by chleibig**
Interesting library!

*Description of changes:*
I was checking out the examples, and found a small but easy to fix issue:

Loading data with prefetching resulted in a runtime error due to a typo inside `PrefetchedGenerator`. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: fortuna/data/loader.py [hunk 1]
```diff
@@ -809,7 +809,7 @@ def __next__(self):
         self._ready = False
         return self._batch
 
-    def prefetch(self):
+    def _prefetch(self):
```
Comment (chleibig) [inline fortuna/data/loader.py:812]: Here's the fix to be in line with the expectation in line 808. I assume the method shall indeed be private to the generator.

File: tests/fortuna/test_data.py [hunk 2]
```diff
@@ -36,6 +35,18 @@ def test_get_inputs_from_generator_function(self):
...
             assert a.shape == (3, 5)
 
+    def test_array_data(self):
+        data_org = make_array_random_data(
+            n_data=10,
+            shape_inputs=(2,),
+            output_dim=1,
+            output_type="continuous",
+        )
+        for prefetch in [False, True]:
```
Comment (chleibig) [inline tests/fortuna/test_data.py:45]: Didn't want to add a new dependency to parametrize the test.

File: tests/fortuna/test_data.py [hunk 3]
```diff
@@ -36,6 +35,18 @@ def test_get_inputs_from_generator_function(self):
             assert type(a) == np.ndarray
             assert a.shape == (3, 5)
 
+    def test_array_data(self):
```
Comment (chleibig) [inline tests/fortuna/test_data.py:38]: Noticed the problem by running an example, but added a test instead.

**Review by chleibig — COMMENTED** [review]
self-review

**Comment by gianlucadetommaso** [comment]
Hi Christian, thanks for this contribution, it's great! You are right, that was definitely a small bug. Thanks for the fix and the unit test. Merging this straightaway.

**Pull request merged**